com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/flowable/RetryProcessAction.java

```diff
@@ -30,20 +30,16 @@ public RetryProcessAction(FlowableFacade flowableFacade, List<AdditionalProcessA
 
     @Override
     protected void executeActualProcessAction(String user, String superProcessInstanceId) {
-        List<String> subProcessIds = getActiveExecutionIds(superProcessInstanceId);
-        ListIterator<String> subProcessesIdsIterator = subProcessIds.listIterator(subProcessIds.size());
-
+        List<String> inactiveExecutionIdsWithoutChildren = flowableFacade.getInactiveExecutionIdsWithoutChildren(superProcessInstanceId);
+        LOGGER.info(MessageFormat.format("Inactive execution ids without children: \"{0}\"", inactiveExecutionIdsWithoutChildren));
         updateUserIfNecessary(user, superProcessInstanceId);
-        while (subProcessesIdsIterator.hasPrevious()) {
```

Comment:
This was done in reverse order because if there are some serviceTasks failed inside a CallActivity, we need to retry first them, then the callActivity and after that the whole process. If you change this -> try to check whether you will not hit any OptimisticLockingException